contributors/design-proposals/auth/pod-restriction.md

```diff
@@ -0,0 +1,477 @@
+	Containers     []ContainerRestriction // Allow any
+
+	RestartPolicy                 *StringRestriction
+	TerminationGracePeriodSeconds *NumericRestriction
+	ActiveDeadlineSeconds         *NumericRestriction
+	DNSPolicy                     *StringRestriction
+	NodeSelector                  *StringMapRestriction
+	AutomountServiceAccountToken  *BoolRestriction
+	SecurityContext               *PodSecurityContextRestriction
+
+    // TODO: Scheduling restrictions are deeply nested structures and require
+    // more thought.
+	// Affinity      *AffinityRestriction
+	// SchedulerName *StringRestriction
+	// Tolerations   []TolerationRestriction
+
+    // ...
+}
+
+type ContainerRestriction struct {
+	Image []ImageRestriction  // Allow Any
+
+	ImagePullPolicy          *StringRestriction
+	SecurityContext          *SecurityContextRestriction
+
+    // ...
+}
+
+// QUESTION: Are image restrictions injecting too much login?
+type ImageRestriction struct {
+	Registry  *StringRestriction
+	Image     *StringRestriction
+	Tag       *StringRestriction
+}
+
+type SecurityContextRestriction struct {
+	Capabilities             *CapabilitiesRestriction
+	Privileged               *BoolRestriction
+	SELinuxOptions           *SELinuxOptionsRestriction
+	RunAsUser                *NumericRestriction
+	RunAsGroup               *NumericRestriction
+	RunAsNonRoot             *BoolRestriction
+	ReadOnlyRootFilesystem   *BoolRestriction
+	AllowPrivilegeEscalation *BoolRestriction
+}
+
+type SELinuxOptionsRestriction struct {
+	PointerRestriction
+	User  *StringRestriction
+	Role  *StringRestriction
+	Type  *StringRestriction
+	Level *StringRestriction
+}
+
+type CapabilitiesRestriction struct {
+	PointerRestriction
+	Add  *StringListRestriction
+	Drop *StringListRestriction
+}
+
+type PodSecurityContextRestriction struct {
+	HostNetwork           *BoolRestriction
+	HostPid               *BoolRestriction
+	HostIPC               *BoolRestriction
+	ShareProcessNamespace *BoolRestriction
+
+    // RunAs* options left to container restrictions.
+	SupplementalGroups    *NumericRestriction
+	FSGroup               *NumericRestriction
+}
+
+type VolumeRestriction struct {
+    CollectionRestriction
+
+    // Mount restrictions apply to the volumes included in this restriction
+	Mount *MountRestriction
+
+	VolumeType *StringListRestriction  // Allowed volume types
+
+    // Type-specific restrictions
+	HostPath *HostPathVolumeRestriction
+    // ...
+}
+
+type MountRestriction struct {
+	ReadOnly         *BoolRestriction
+	Mountpropagation *StringRestriction
+}
+
+type HostPathVolumeRestriction struct {
+	Path *StringRestriction // Prefix match whitelist / blacklist
+	Type *StringRestriction
+}
+
+type PointerRestriction struct {
+	ForbidNil  bool
+	RequireNil bool
+}
+
+type CollectionRestriction struct {
+	ForbidEmpty  bool
+	RequireEmpty bool
+}
+
+type StringRestriction struct {
+	PointerRestriction
+
+	Whitelist []string
```

Comment:
Does this mean include/exclude, or accept/deny?

What's the rationale for per-field accept/deny rather than top-level accept/deny?